include/nbl/builtin/hlsl/prefix_sum_blur.hlsl

```diff
@@ -0,0 +1,262 @@
+                const T ceiled = prefixSumAccessor.template get<T, uint32_t>(rightClIdx % linearSize) + ceil(float32_t(rightClIdx % lastIdx) / linearSize) * last;
+                result += lerp(floored, ceiled, alpha);
+                break;
+            }
+            case ETC_CLAMP_TO_BORDER:
+            {
+                result += prefixSumAccessor.template get<T, uint32_t>(lastIdx) + (rightIdx - lastIdx) * borderColor;
+                break;
+            }
+            case ETC_CLAMP_TO_EDGE:
+            {
+                const T last = prefixSumAccessor.template get<T, uint32_t>(lastIdx);
+                const T lastMinusOne = prefixSumAccessor.template get<T, uint32_t>(lastIdx - 1);
+                result += (rightIdx - lastIdx) * (last - lastMinusOne) + last;
+                break;
+            }
+            case ETC_MIRROR:
+            {
+                const T last = prefixSumAccessor.template get<T, uint32_t>(lastIdx);
+                T floored, ceiled;
+                int32_t d = rightFlIdx - lastIdx;
+
+                if (d % (2 * linearSize) == linearSize)
+                    floored = ((d + linearSize) / linearSize) * last;
+                else
+                {
+                    const uint32_t period = uint32_t(ceil(float32_t(d) / linearSize));
+                    if ((period & 0x1u) == 1)
+                        floored = period * last + last - prefixSumAccessor.template get<T, uint32_t>(lastIdx - uint32_t(d % linearSize));
+                    else
+                        floored = period * last + prefixSumAccessor.template get<T, uint32_t>((d - 1) % linearSize);
+                }
+                
+                d = rightClIdx - lastIdx;
+                if (d % (2 * linearSize) == linearSize)
+                    ceiled = ((d + linearSize) / linearSize) * last;
+                else
+                {
+                    const uint32_t period = uint32_t(ceil(float32_t(d) / linearSize));
+                    if ((period & 0x1u) == 1)
+                        ceiled = period * last + last - prefixSumAccessor.template get<T, uint32_t>(lastIdx - uint32_t(d % linearSize));
+                    else
+                        ceiled = period * last + prefixSumAccessor.template get<T, uint32_t>((d - 1) % linearSize);
+                }
+
+                result += lerp(floored, ceiled, alpha);
+                break;
+            }
+            case ETC_MIRROR_CLAMP_TO_EDGE:
+            {
+                const T last = prefixSumAccessor.template get<T, uint32_t>(lastIdx);
+                const T first = prefixSumAccessor.template get<T, uint32_t>(0);
+                const T firstPlusOne = prefixSumAccessor.template get<T, uint32_t>(1);
+                result += (rightIdx - lastIdx) * (firstPlusOne - first) + last;
+                break;
+            }

```

Comment:
I'm gonna accept on blind faith these work, don't have time to review, please double check or get someone else to play devil's advocate

Reply:
@Przemog1 could be appropriate person to check the math here

I've spotted one wrong thing already, since your prefix sum is inclusive, `first` is already the value you want, not `firstPlusOne-first` for clamp to edge. Also you don't handle the first mirror correctly.

Btw now that this is separate from the shader, you can simple-test this box sampler on the CPU side with a simple array of `{1,2,3,4,5}` and its prefix sum